e2e/cypress/integration/messaging/collapse_link_spec.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        // # Set the configuration on Link Previews
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                EnableLinkPreviews: true,
+            },
+        });
+
+        cy.apiLogin('user-1');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
+
+        cy.apiLogin('sysadmin');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
+
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18708-Link preview - Removing it from my view removes it from other user\'s view', () => {
+        const message = 'https://www.bbc.com/news/uk-wales-45142614';
+
+        // # Create new DM channel with user's email
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Post message to use
+        cy.postMessage(message);
+
+        cy.getLastPostId().then((postId) => {
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
+
+            // # Log-in to the other user
+            cy.apiLogin('sysadmin');
+            cy.visit('/ad-1/channels/town-square');
+
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
+
+            // # Log-in back to the first user
+            cy.apiLogin('user-1');
+            cy.visit('/ad-1/channels/town-square');
+
+            // # Collapse the preview
+            cy.get(`#${postId}_message`).find('.post__embed-visibility').click({force: true});
+
+            // * Check the preview is not shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('not.exist');
+
+            // # Log-in to the other user
+            cy.apiLogin('sysadmin');
+            cy.visit('/ad-1/channels/town-square');
+
+            // * Check the preview is shown
+            cy.get(`#${postId}_message`).find('.file-preview__button').should('exist');
```

Comment:
The testcase fails for me in this test step. I tried to reproduce this issue manually, but for a sysadmin the preview is displayed even if user-1 collapses the preview. So something is wrong with the Cypress test which is causing this issue. 

![Screenshot 2019-11-11 at 10 27 28 PM](https://user-images.githubusercontent.com/1429138/68606103-b769db80-04d3-11ea-83aa-c7af8f5d777f.png)

Reply:
Just realised this might have failed because of an issue in the application. A PR for this has already been submitted by @saturninoabril 
https://github.com/mattermost/mattermost-webapp/pull/4177